sycl/include/sycl/ext/oneapi/experimental/enqueue_functions.hpp

```diff
@@ -99,15 +99,26 @@ template <typename LCRangeT, typename LCPropertiesT> struct LaunchConfigAccess {
 template <typename CommandGroupFunc, typename PropertiesT>
 void submit_impl(queue &Q, PropertiesT Props, CommandGroupFunc &&CGF,
                  const sycl::detail::code_location &CodeLoc) {
-  Q.submit_without_event(Props, std::forward<CommandGroupFunc>(CGF), CodeLoc);
+  Q.submit_without_event<
+#if __SYCL_USE_FALLBACK_ASSERT
+      true
+#else
+      false
+#endif
+      >(Props, detail::type_erased_cgfo_ty{CGF}, CodeLoc);
```

Comment:
Looks like we always have `__SYCL_USE_FALLBACK_ASSERT` set to a value, could we not just use it directly here?

```suggestion
  Q.submit_without_event<__SYCL_USE_FALLBACK_ASSERT>(Props, detail::type_erased_cgfo_ty{CGF}, CodeLoc);
```